### DISCUSSION TITLE
Hosting 'static' Next.JS apps from BFFs written in another language

### DISCUSSION THREAD

**Discussion by tobysmith568** [vercel/next.js#71976] [Help]
### Summary

Hey,

I've been writing a proof of concept app recently where we're trying to pair a 'static' Next.JS app with a [Backend for Frontend](https://samnewman.io/patterns/architectural/bff/) server app (BFF) written in another language (in our case it's C#). We're also using the pages directory rather than the app directory if that's relevant.

I use the word static above in quotes because while we've not been using `getServersideProps`, we have been using dynamic routes (`/company/[companyId]/users/[userId]`) _without_ using `getStaticPaths`. With `output: "standalone"` in our next.config.js this seems to create an interesting build output where under `server/pages` we see files like `company/[companyId]/users/[userId].html` and `company/[companyId]/users/[userId].js`.

My question is: if we wrote middleware in our BFF that, on intercepting a 404 being returned from itself, tried to match the request with one of these html/js files, would that be sufficient for serving the Next.JS app as a part of the BFF? (Assuming it also served the `/static` build output directory for CSS etc.?)

- Would this work?
  - If yes, is this all that the JavaScript server that's bundled with Next.JS is doing provided you don't use Next.JS-specific features like `getServersideProps`?
  - If yes, how likely is it that this would break in the future?

Thanks for any input!

**Comment by aniruddhaadak80** [comment]
Hi @tobysmith568 ,

Your approach to integrating a 'static' Next.js app with a BFF in C# sounds interesting! 

To address your questions:

1. **Serving Files on 404**: Yes, intercepting 404 responses in your BFF and matching them with the corresponding HTML/JS files from your Next.js build output could work for serving your Next.js app. Just ensure that your BFF can also handle the static assets (CSS, etc.) from the output directory.

2. **Functionality of the Next.js Server**: Essentially, the JavaScript server bundled with Next.js handles similar routing for static files, serving them based on the requests it receives. If you're not using features like `getServerSideProps`, your approach should align with its functioning.


Thanks!
